barretenberg/cpp/src/barretenberg/vm2/simulation/execution.cpp

```diff
@@ -151,6 +151,27 @@ void Execution::jumpi(ContextInterface& context, MemoryAddress cond_addr, uint32
     }
 }
 
+void Execution::internal_call(ContextInterface& context, uint32_t loc)
+{
+
+    auto& internal_call_stack_manager = context.get_internal_call_stack_manager();
+    // todo: range check loc
```

Comment:
Not needed? It's an immediate and from instr fetching (well the spec) you know it cannot be > 32 bits IIRC.

You can add an assert though, but it shouldn't fail.